src/Dictionaries/DirectDictionary.cpp

```diff
@@ -426,7 +426,11 @@ void DirectDictionary::getItemsImpl(
     for (const auto row : ext::range(0, rows))
         value_by_key[ids[row]] = get_default(row);
 
-    auto stream = source_ptr->loadAll();
+    std::vector<Key> to_load;
```

Comment:
Please also add reserve, because we know the number of distinct keys exactly (as hash table size).